lib/client.js

```diff
@@ -40,19 +43,38 @@ function* createClient(rabbitmqUrl, options) {
     consume,
     listen,
     publish,
-    close: co.wrap(function* close() {
+    close: co.wrap(function* close(forceClose = false) {
       if (this.channel !== null) {
         yield this.channel.close();
         this.channel = null;
       }
       if (this.connection !== null) {
+        this.connection.removeListener('close', handleExitOnConnectionClose);
         yield this.connection.close();
         this.connection = null;
       }
+
+      if (forceClose) {
+        yield cb => setTimeout(cb, options.processExitTimeout);
+        process.exit(1);
```

Comment:
This library will exit the wall node process.
Better notify the library user, and exit the process depending on the use-case.

Reply:
As discussed, this is specifically used when handling the event, and is a first step to handle it.

But yes, I'll create a ticket to send an event to the host app, so it can handle it.